### ISSUE TITLE
Bookmarks order is wrong (again?) / Disabled floccus takes precedence when syncing back

### ISSUE THREAD

**Issue by Neurology0443** [floccusaddon/floccus#1752]
### Which version of floccus are you using?

5.3.1

### How many bookmarks do you have, roughly?

30

### Are you using other means to sync bookmarks in parallel to floccus?

No

### Sync method

WebDAV

### Which browser are you using? In case you are using the phone App, specify the Android or iOS version and device please.

Firefox 128 (64bit)

### Which version of Nextcloud Bookmarks are you using? (if relevant)

/

### Which version of Nextcloud? (if relevant)

/

### What kind of WebDAV server are you using? (if relevant)

Webdav from SFTPGO as docker container.

### Describe the Bug

Issue 1: After manual or automatic sync, my bookmarks and folders are appended and not in the correct order. However this does not happened everytime.

Issue 2: If profile's 2 floccus is disabled and I delete all bookmarks and re-enable floccus, profile 2 takes precedence over profile 1 even if sync is manually triggered on profile 1 before profile 2.

### Expected Behavior

Issue 1: Folders and bookmarks should have the same order in each browser

Issue 2: Floccus from profile 1 should take precedence over profile 2 (because Floccus was disable on profile 2)

### To Reproduce

Sync method: Webdav (docker sftpgo)
Browser: Firefox Version 128.0 (64-bit)
Testing environment: MacOS Sonoma 14.7

2 new Firefox profiles:

- profile 1
- profile 2
- Delete all default bookmarks/folder from both profiles

Floccus installation on both profiles:

- create new floccus profile = floccusDebug
- connect to webdav share with credentials
- create new bookmarks.xbel = floccusDebug.xbel
- bookmarks folder = Bookmarks Toolbar

Floccus sync configuration:

- Auto-sync
- 15 min sync interval
- Always merge local changes with changes from other browsers
- Include this profile's folder in other profiles

Bookmarks/folder annotation:

Folder X.Y
Bookmark X.Y

X = profile 1 or profile 2
Y = nth folder or bookmark 

### 1. Manual sync
---

**Issue 1.**

- Create a few new dummy folders/bookmarks in profile 1
	- folder 1.1; folder 1.2
	- bookmark 1.1; bookmark 1.2; bookmark 1.3

- Sync floccus now manually on profile 1
- Sync floccus now manually on profile 2

Profile 1
<img width="1440" alt="image" src="https://github.com/user-attachments/assets/5c3fa7af-5241-48b8-8fa2-a4bb1ca55ef8">

Profile 2
<img width="1440" alt="image" src="https://github.com/user-attachments/assets/ee67bd80-65e1-450d-a406-1bfad1b18939">

**_However_**, If I manually trigger on profile 2 with the synchronization strategy `Always undo local changes and download changes from other browsers` it gets correctly reordered

Profile 2
<img width="1440" alt="image" src="https://github.com/user-attachments/assets/22669e0c-04e3-405a-a093-5a2622fff4c1">

**Issue 2.**

- Disable floccus addon on Firefox profile 2
- Delete all bookmarks and folder on Firefox profile 2
- Enable floccus addons on Firefox profile 2
- Sync floccus now manually on profile 1
- Sync floccus now manually on profile 2
- Sync floccus now mannully on profile 1

Profile 1
<img width="1440" alt="IMG-241022155830" src="https://github.com/user-attachments/assets/c9da5130-3516-4803-920a-adb97f95d4ef">

Profile 2
<img width="1440" alt="IMG-241022155732" src="https://github.com/user-attachments/assets/a74017c2-3db6-4a2f-a4fe-27777f96ca06">

~~**_However_**, If I manually trigger on profile 2 with the synchronization strategy `Always undo local changes and download changes from other browsers` this bug disappears and is not reproducible anymore. Best way to test it is to remove the plugin and reinstall it.~~

### 2. Auto sync
---
Synchronization interval = 5min

Same *steps as Issues 1. Manual sync* except the **Floccus auto-sync** kicks in.
Exact same results as **manual sync**.

Same *steps as Issue 2. Manual sync* except the **Floccus auto-sync** kicks in.
Exact same results as **manual sync**.

### Debug logs

Debug logs issue 1

Profile 1
[floccus-5.3.1-2024-10-22-redacted-issue1-profile1.log](https://github.com/user-attachments/files/17479159/floccus-5.3.1-2024-10-22-redacted-issue1-profile1.log)

Profile 2
[floccus-5.3.1-2024-10-22-redacted-issue1-profile2.log](https://github.com/user-attachments/files/17479161/floccus-5.3.1-2024-10-22-redacted-issue1-profile2.log)
[floccus-5.3.1-2024-10-22-redacted-issue1-profile2-syncmethod.log](https://github.com/user-attachments/files/17479162/floccus-5.3.1-2024-10-22-redacted-issue1-profile2-syncmethod.log)

Debug logs issue 2

Profile 1
[floccus-5.3.1-2024-10-22-redacted-issue2-profile1.log](https://github.com/user-attachments/files/17479334/floccus-5.3.1-2024-10-22-redacted-issue2-profile1.log)

Profile 2
[floccus-5.3.1-2024-10-22-redacted-issue2-profile2.log](https://github.com/user-attachments/files/17479339/floccus-5.3.1-2024-10-22-redacted-issue2-profile2.log)

### Debug log provided

- [X] I have provided a debug log file

**Comment by marcelklehr** [comment]
Thank you for the detailed reproduction steps :blue_heart: I'll try to look into this soon! Sorting issues are a bit lower on my priority list at the moment, though.

**Comment by marcelklehr** [comment]
> Was there any update regarding your extension ?

Yes, there have been quite a few updates, but I haven't explicitly tried to fix this issue 🤔  If it works for you now, I'm happy to close. If it should ever come back, we can always reopen :)
